seq_tools/validation/c691_check_strand_specificity_in_fastq.py

```diff
@@ -0,0 +1,230 @@
+                self.message = message
+                self.status = 'INVALID'
+                problematic_fastqs.append(fastq)
+            else:
+                message =  "Strand orientation for RNA-Seq file %s detected as %s. Matching metadata %s : PASS"  % (fastq,strand_orientation,self.metadata.get('experiment')['library_strandedness'])
+                self.logger.info(f'[{self.checker}] {message}')
+                self.message = message
+                                                                                                        
+        if len(problematic_fastqs)>0:
+            self.status = 'INVALID'
+            message = "The following RNA-seq FASTQs' stand orientation does not match metadata : %s" % ",".join(problematic_fastqs)                
+            self.message = message
+            self.logger.info(f'[{self.checker}] {message}')
+            return
+        else:
+            self.status = 'PASS'
+            message = "All FASTQs strand orientation check : PASS"
+            self.message = message
+            self.logger.info(f'[{self.checker}] {message}')
+            return
+
+def determine_strandedness(salmon_json,is_paired_end):
+    with open(salmon_json, 'r') as f:
+        data=f.read()
+        salmon_dict = json.loads(data)
+
+    convert_terms={
+        "IU":"UNSTRANDED",
+        "U":"UNSTRANDED",
+        "ISR":"FIRST_READ_SENSE_STRAND",
+        "SR":"FIRST_READ_SENSE_STRAND",
+        "ISF":"FIRST_READ_ANTISENSE_STRAND",
+        "SF":"FIRST_READ_ANTISENSE_STRAND",
+    }
+    print(convert_terms[salmon_dict['library_types'][0]])
+    return convert_terms[salmon_dict['library_types'][0]]
+
+def fastq_test_length(fastq):
+    if fastq.endswith("fastq.gz") or fastq.endswith("fq.gz"):
+        cmd="cat "+fastq+" | zcat | wc -l"
+    else:
+        cmd="cat "+fastq+" | bzcat | wc -l"
+
+    count_cmd = subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
+
+    file_size=int(count_cmd.decode('utf-8').strip())/ 4
+    
+    return file_size
+
+def align_fastq(read1,read2):
+    if read2:
+        cmd="salmon quant "\
+            "-i tmp/transcript_index "\
+            "-l A "\
+            "-1 "+read1+" "\
+            "-2 "+read2+" "\
+            "--validateMappings "\
+            "-o transcript_quant"
+    else:
+        cmd="salmon quant "\
+            "-i tmp/transcript_index "\
+            "-l A "\
+            "-1 "+read1+" "\
+            "--validateMappings "\
+            "-o transcript_quant"
+
+    subprocess.check_output(cmd,stderr=subprocess.STDOUT,shell=True)
+    return "transcript_quant/aux_info/meta_info.json"
+
+def split_fastq(read1,read2,num):
+    if read1.endswith("gz"):
+        cmd="cat "+read1+"| zcat | head -n "+str(0.25*num)+" | gzip > tmp/tmp_"+str(num)+"_read1.fastq.gz"
+    elif read2 and read2.endswith("bz2"):
+        cmd="cat "+read1+"| bzcat | head -n "+str(0.25*num)+" | gzip > tmp/tmp_"+str(num)+"_read1.fastq.gz"
```

Comment:
Typos in `elif read2 and read2.endswith("bz2"):`
should be `elif read1 and read1.endswith("bz2"):`